### PR TITLE
Add backward compatibility support for private customer plugins

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR adds backward compatiblity support for private customer plugins

Summary:
- This is to support a case where, when a user using CLI prior to Central repository support, used CLIPlugin CR to download the custom private plugins hosted on their private repository. This is just to provide the backward compatibility and not for general usage.
- User should use the environment variable `TANZU_CLI_PRIVATE_PLUGIN_DISCOVERY_IMAGES` to specify the custom private repository where he hosted his custom plugins. Users who are currently using the CLIPlugin CR to point to their cutom private plugin repository, should use the CLI documentation to build and publish their plugins to their private repository so as to use this approach.
- The newly introduced environment variable is same as `TANZU_CLI_ADDITIONAL_PLUGIN_DISCOVERY_IMAGES_TEST_ONLY` except the name change and the new environment `TANZU_CLI_PRIVATE_PLUGIN_DISCOVERY_IMAGES`variable would be mutual exclusive to `TANZU_CLI_ADDITIONAL_PLUGIN_DISCOVERY_IMAGES_TEST_ONLY` (only one of them would be honored) with `TANZU_CLI_PRIVATE_PLUGIN_DISCOVERY_IMAGES` taking the priority.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Ran the tanzu plugin search and it shows only the `test` and `build` plugin from default plugin source
```
❯ ./bin/tanzu plugin search
  NAME     DESCRIPTION             TARGET  LATEST
  builder  Build Tanzu components  global  v0.90.0-alpha.1
  test     Test the CLI            global  v0.90.0-alpha.1
❯ ./bin/tanzu plugin source list
  NAME     IMAGE
  default  projects.registry.vmware.com/tanzu_cli_stage/plugins/plugin-inventory:latest
❯
❯

```
Exported  the new environment variable `TANZU_CLI_PRIVATE_PLUGIN_DISCOVERY_IMAGES` by pointing  it to the additional repositories and `tanzu plugin search` command shows the additional plugins as well. 
```
❯ tanzu config set env.TANZU_CLI_PRIVATE_PLUGIN_DISCOVERY_IMAGES localhost:9876/tanzu-cli/plugins/sandbox1:small
❯
❯
❯ ./bin/tanzu plugin search
  NAME                DESCRIPTION                  TARGET           LATEST
  builder             Build Tanzu components       global           v0.90.0-alpha.1
  test                Test the CLI                 global           v0.90.0-alpha.1
  cluster             Desc for cluster             kubernetes       v11.11.11
  feature             Desc for feature             kubernetes       v11.11.11
  kubernetes-release  Desc for kubernetes-release  kubernetes       v11.11.11
  management-cluster  Desc for management-cluster  kubernetes       v11.11.11
  package             Desc for package             kubernetes       v11.11.11
  secret              Desc for secret              kubernetes       v11.11.11
  telemetry           Desc for telemetry           kubernetes       v11.11.11
  account             Desc for account             mission-control  v11.11.11
  apply               Desc for apply               mission-control  v11.11.11
  audit               Desc for audit               mission-control  v11.11.11
  cluster             Desc for cluster             mission-control  v11.11.11
  clustergroup        Desc for clustergroup        mission-control  v11.11.11
  continuousdelivery  Desc for continuousdelivery  mission-control  v11.11.11
  data-protection     Desc for data-protection     mission-control  v11.11.11
  ekscluster          Desc for ekscluster          mission-control  v11.11.11
  events              Desc for events              mission-control  v11.11.11
  helm                Desc for helm                mission-control  v11.11.11
  iam                 Desc for iam                 mission-control  v11.11.11
  inspection          Desc for inspection          mission-control  v11.11.11
  integration         Desc for integration         mission-control  v11.11.11
  management-cluster  Desc for management-cluster  mission-control  v11.11.11
  policy              Desc for policy              mission-control  v11.11.11
  secret              Desc for secret              mission-control  v11.11.11
  tanzupackage        Desc for tanzupackage        mission-control  v11.11.11
  workspace           Desc for workspace           mission-control  v11.11.11
```

Installing the plugin from the additional plugin source was successful
```
❯ ./bin/tanzu plugin install feature
[i] Installing plugin 'feature:v11.11.11' with target 'kubernetes'
[ok] successfully installed 'feature' plugin
```

### Release note

```release-note
Add backward compatibility support for private customer plugins
```

### CODE REVIEW

File: pkg/pluginmanager/manager.go [hunk 1]
```diff
@@ -209,7 +209,7 @@ func getAdditionalTestPluginDiscoveries() []configtypes.PluginDiscovery {
 	for idx, image := range testDiscoveryImages {
 		testDiscoveries = append(testDiscoveries, configtypes.PluginDiscovery{
 			OCI: &configtypes.OCIDiscovery{
-				Name:  fmt.Sprintf("test_%d", idx),
```
Comment (prkalle) [inline pkg/pluginmanager/manager.go:212]: self: Modified the OCI discovery source name  from `test_` to `disc_` to be generic. Would like to hear opinions on this.

**Review by marckhouzam — APPROVED** [review]
Thanks @prkalle.
LGTM

File: pkg/config/defaults_test.go [hunk 2]
```diff
@@ -109,5 +113,41 @@ var _ = Describe("defaults test cases", func() {
...
+
+			oldPrivateDiscValue := os.Getenv(constants.ConfigVariableAdditionalPrivateDiscoveryImages)
+			err = os.Setenv(constants.ConfigVariableAdditionalPrivateDiscoveryImages,
+				privateHost1+"/private/path, "+privateHost2+"/another/private/image")
+			Expect(err).To(BeNil())
+
+			trustedRegis := GetTrustedRegistries()
+			Expect(trustedRegis).NotTo(BeNil())
+			Expect(trustedRegis).Should(ContainElement(privateHost1))
+			Expect(trustedRegis).Should(ContainElement(privateHost2))
```
[the remark this answers is not in the document]
Comment (prkalle) [reply pkg/config/defaults_test.go:144]: updated. Thanks!

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks for adding this.

**Pull request merged**